### PR TITLE
Bug fix: don't set non-conditional ruleset when existing ruleset already references the generated ruleset

### PR DESCRIPTION
**Pull request by rita-g-sonarsource**
Caused by #863 and #805

### CODE REVIEW

File: src/Integration/Binding/CSharpVBBindingOperation.cs [hunk 1]
```diff
@@ -153,7 +153,9 @@ private void AddRuleset()
         /// </summary>
         private void TrySetNonConditionalRuleSet()
         {
-            var hasUserSpecifiedRuleSet = propertyInformationMap.Keys.Any(x => !IsDefaultMicrosoftRuleSet(x.Value as string));
```
Comment (rita-g-sonarsource) [inline src/Integration/Binding/CSharpVBBindingOperation.cs:156]: The bug was that in the `CalculateRuleSetInformation` method I ignore all rulesets that already reference the generated ruleset.
So propertyInformationMap was empty.

File: src/Integration/Binding/CSharpVBBindingOperation.cs [hunk 2]
```diff
@@ -153,7 +153,9 @@ private void AddRuleset()
...
+            var solutionRuleSetProvider = serviceProvider.GetService<ISolutionRuleSetsInformationProvider>();
+            var ruleSetsInfo = solutionRuleSetProvider.GetProjectRuleSetsDeclarations(initializedProject).ToArray();
+            var hasUserSpecifiedRuleSet = ruleSetsInfo.Any(x => !IsDefaultMicrosoftRuleSet(x.RuleSetPath));
```
Comment (duncanp-sonar) [inline src/Integration/Binding/CSharpVBBindingOperation.cs:158]: This doesn't feel quite right: there's quite an overlap between this processing and what happens in `CalculateRuleSetInformation`.
`CalculateRuleSetInformation` which has already fetched the `rulesetsInfo` so calculating the `hasUserSpecifiedRuleSet` there would be a one-liner.Also, the naming of `CalculateRuleSetInformation` suggests it's intended to be the place were the calculations are done.

Comment (rita-g-sonarsource) [reply src/Integration/Binding/CSharpVBBindingOperation.cs:158]: So you're suggesting to extract `hasUserSpecifiedRuleSet` to a member?

Comment (duncanp-sonar) [reply src/Integration/Binding/CSharpVBBindingOperation.cs:158]: Yep. Logically it's the same kind of data as `propertyInformationMap`.

**Pull request merged**